webpack/assets/javascripts/react_app/redux/API/APIRequest.js

```diff
@@ -1,24 +1,45 @@
 import { API } from './';
+import { actionTypeGenerator } from './APIActionTypeGenerator';
+import {
+  formatSuccess as defaultFormatSuccess,
+  formatFailure as defaultFormatFailure,
+  formatPending as defaultFormatPending,
+} from './APIHelpers';
 
-export const get = async (payload, url, store, actionTypes) => {
-  store.dispatch({
-    type: actionTypes.REQUEST,
+export const get = async (action, store) => {
+  const {
+    key,
+    url,
     payload,
+    actionTypes,
+    formatPending = defaultFormatPending,
+    formatSuccess = defaultFormatSuccess,
+    formatFailure = defaultFormatFailure,
+  } = action;
+  const { REQUEST, SUCCESS, FAILURE } = actionTypeGenerator(key, actionTypes);
+
+  store.dispatch({
+    key,
+    type: REQUEST,
+    payload: formatPending({ payload }),
   });
+
   try {
     const { data } = await API.get(
       url,
       payload.headers || {},
       payload.params || {}
     );
     store.dispatch({
-      type: actionTypes.SUCCESS,
-      payload: { ...payload, ...data },
+      key,
+      type: SUCCESS,
+      payload: formatSuccess({ data, payload }),
     });
   } catch (error) {
     store.dispatch({
-      type: actionTypes.FAILURE,
-      payload: { ...payload, error },
+      key,
+      type: FAILURE,
+      payload: formatFailure({ error, payload }),
```

Comment:
Same here, I would expect to have `error` and `payload` separate.

Reply:
same response as in here: https://github.com/theforeman/foreman/pull/7222#discussion_r360788810